src/main/java/org/jitsi/jicofo/conference/Participant.java

```diff
@@ -143,6 +151,49 @@ public Participant(
         );
     }
 
+    /**
+     * Notify this {@link Participant} that the underlying {@link ChatRoomMember}'s presence changed.
+     */
+    void presenceChanged()
+    {
+        if (updateDesktopSourceIsMuted(roomMember.getSourceInfos()))
+        {
+            conference.desktopSourceIsMutedChanged(this, desktopSourceIsMuted);
+        }
+    }
+
+    /**
+     * Update the value of {@link #desktopSourceIsMuted} based on the advertised {@link SourceInfo}s.
+     * @return true if the value of {@link #desktopSourceIsMuted} changed as a result of this call.
+     */
+    private boolean updateDesktopSourceIsMuted(@NotNull Set<SourceInfo> sourceInfos)
+    {
+        boolean newValue = sourceInfos.stream().anyMatch(si -> si.getVideoType() == VideoType.Desktop && si.getMuted());
+        if (desktopSourceIsMuted != newValue)
+        {
+            desktopSourceIsMuted = newValue;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Notify this participant that another participant's (identified by {@code owner}) screensharing source was muted
+     * or unmuted.
+     */
+    void remoteDesktopSourceIsMutedChanged(Jid owner, Boolean muted)
+    {
+        // This is only needed for backwards compatibility with clients that don't support receiving multiple streams.
+        if (supportsReceivingMultipleVideoStreams())
+        {
+            return;
+        }
+
+        sourceSignaling.remoteDesktopSourceIsMutedChanged(owner, muted);
+        // Signal updates, if any, immediately.
+        scheduleSignalingOfQueuedSources(0);
```

Comment:
Any worries if this mixes with lots of participants joining at once, i.e. the case for which we originally queued the sources?